datadog/resource_datadog_synthetics_test_.go

```diff
@@ -111,7 +111,44 @@ func syntheticsTestRequest() *schema.Schema {
 
 func syntheticsTestOptions() *schema.Schema {
 	return &schema.Schema{
-		Type:     schema.TypeMap,
+		Type: schema.TypeMap,
+		DiffSuppressFunc: func(key, old, new string, d *schema.ResourceData) bool {
+			if key == "options.follow_redirects" {
+				// TF nested schemas is limited to string values only
+				// follow_redirects being a boolean in Datadog json api
+				// we need a sane way to convert from boolean to string
+				// and from string to boolean
+				truthyValues := map[string]struct{}{"true": {}, "1": {}}
+				falsyValues := map[string]struct{}{"false": {}, "0": {}}
+				_, isTruthyOld := truthyValues[old]
+				_, isFalsyOld := falsyValues[old]
+				_, isTruthyNew := truthyValues[new]
+				_, isFalsyNew := falsyValues[new]
+				if !isTruthyOld && !isFalsyOld {
+					isFalsyOld = true
+				}
+				if !isTruthyNew && !isFalsyNew {
+					isFalsyNew = true
+				}
+				return (isTruthyNew && isTruthyOld) || (isFalsyNew && isFalsyOld)
+			}
+			return old == new
+		},
+		ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+			expectedValues := map[string]struct{}{"true": {}, "false": {}, "1": {}, "0": {}}
```

Comment:
Could we use parseBool here too - https://golang.org/pkg/strconv/#ParseBool
If that function returns doesn't return an error then this validation passes?

Reply:
`ParseBool` does not fit my use case here, but I rewrote it with a switch _à la_ `ParseBool`